File: FactorySimulation/src/factory/Unit.java
package factory;


import java.util.List;

public class Unit {
	
	
	List<Machine> machines;
	Unit nextUnit;
	boolean lastUnit;
	
	
	public void update() {
		
		if(machines.size() ==1) {
			Machine m = machines.get(0);
			
			if(m.getStatus() ==1) {
				if(!lastUnit && nextUnit.getReady()) {
					
					if(m.getWorkingOn() <=0) {
						m.setStatus(0);
					}else {
						nextUnit.pushProduct(m.getWorkingOn());
						m.setStatus(0);
					}
				
				}else if(lastUnit) {
					m.setStatus(0);
				}
			}
			
			m.update();
			
			
		}else {
			Machine m1 = machines.get(0);
			Machine m2 = machines.get(1);
			
			if(m1.getStatus() ==1 && m2.getStatus() ==1) {
				if(!lastUnit && nextUnit.getReady()) {
					nextUnit.pushProduct(m1.getWorkingOn());
					m1.setStatus(0);
					m2.setStatus(0);
					
					
				}else if(lastUnit) {
					m1.setStatus(0);
					m2.setStatus(0);
				}
			}
			
			
			m1.update();
			m2.update();
			
			
		}
		
		
	
		
	}


	public List<Machine> getMachines() {
		return machines;
	}


	public void setMachines(List<Machine> machines) {
		this.machines = machines;
	}
	
	public void pushProduct(int productId) {
		for(Machine m : machines) {
			m.setWorkingOn(productId);
			
		}
	}
	
	public boolean getReady() {
		
		for(Machine m : machines) {
			if(m.getStatus() !=0) {
				return false;
			}
			
		}
		return true;
	}


	public Unit getNextUnit() {
		return nextUnit;
	}


	public void setNextUnit(Unit nextUnit) {
		this.nextUnit = nextUnit;
	}


	public boolean isLastUnit() {
		return lastUnit;
	}


	public void setLastUnit(boolean lastUnit) {
		this.lastUnit = lastUnit;
	}
	

}

File: FactorySimulation/C# Simulation/FactorySimulation/Unit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FactorySimulation
{
    class Unit
    {


		private List<Machine> _machines;
		private Unit _nextUnit;
		private bool _lastUnit;


		public void Update()
		{

			if (_machines.Count == 1)
			{
				Machine m = _machines[0];

				if (m.GetStatus() == 1)
				{
					if (!_lastUnit && _nextUnit.GetReady())
					{

						if (m.GetWorkingOn() <= 0)
						{
							m.SetStatus(0);
						}
						else
						{
							_nextUnit.PushProduct(m.GetWorkingOn());
							m.SetStatus(0);
						}

					}
					else if (_lastUnit)
					{
						m.SetStatus(0);
					}
				}

				m.Update();


			}
			else
			{
				Machine m1 = _machines[0];
				Machine m2 = _machines[1];

				if (m1.GetStatus() == 1 && m2.GetStatus() == 1)
				{
					if (!_lastUnit && _nextUnit.GetReady())
					{
						_nextUnit.PushProduct(m1.GetWorkingOn());
						m1.SetStatus(0);
						m2.SetStatus(0);


					}
					else if (_lastUnit)
					{
						m1.SetStatus(0);
						m2.SetStatus(0);
					}
				}


				m1.Update();
				m2.Update();


			}




		}


		public List<Machine> GetMachines()
		{
			return _machines;
		}


		public void SetMachines(List<Machine> machines)
		{
			this._machines = machines;
		}

		public void PushProduct(int productId)
		{
			foreach(Machine m in _machines)
			{
				m.SetWorkingOn(productId);

			}
		}

		public bool GetReady()
		{

			foreach(Machine m in _machines)
			{
				if (m.GetStatus() != 0)
				{
					return false;
				}

			}
			return true;
		}


		public Unit GetNextUnit()
		{
			return _nextUnit;
		}


		public void SetNextUnit(Unit nextUnit)
		{
			this._nextUnit = nextUnit;
		}


		public bool IsLastUnit()
		{
			return _lastUnit;
		}


		public void SetLastUnit(bool lastUnit)
		{
			this._lastUnit = lastUnit;
		}


	}
}
